packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -1081,48 +1089,65 @@ export class ProjectService extends BaseService {
             );
         }
 
-        // If requireUserCredentials is true, we need to override the existing credentials with the user credentials
-        // even if we use organization credentials
-        if (credentials.requireUserCredentials) {
-            if (!isRegisteredUser) {
-                throw new ForbiddenError(
-                    'Embedded users cannot use personal warehouse credentials',
-                );
-            }
+        // Check if user has their own credentials for this project's warehouse type
+        // Only fetch user credentials when:
+        // 1. requireUserCredentials is enabled (user credentials are mandatory)
+        // 2. Databricks warehouse (supports optional user OAuth credentials)
+        const shouldFetchUserCredentials =
```

Comment:
This is a behaviour change for user warehouse credentials. 

I'd like to test it first with the customer before enabling this feature for other warehouses.